event_routing_backends/processors/mixins/base_transformer.py

```diff
@@ -102,3 +102,16 @@ def transform(self):
                 )
 
         return self.transformed_event
+
+    def extract_username(self):
+        """
+        Extracts username from event by finding it in context first and falling back to data
+        if context does have username key
+
+        Returns:
+            str
+        """
+
+        return self.event['data']['username'] \
```

Comment:
nit: I think in order to have protection against `KeyError` we do something like this
```
username = self.event.get('context', {}).get('username')
if not username:
    username = self.event.get('data', {}).get('username')
return username
```

Reply:
done